modules/ingest-geoip/src/main/java/org/elasticsearch/ingest/geoip/GeoIpDownloaderTaskExecutor.java

```diff
@@ -154,19 +165,30 @@ private void startTask(Runnable onFailure) {
     }
 
     private void stopTask(Runnable onFailure) {
-        ActionListener<PersistentTasksCustomMetadata.PersistentTask<?>> listener = ActionListener.wrap(r -> {}, e -> {
-            if (e instanceof ResourceNotFoundException == false) {
-                logger.error("failed to remove geoip downloader task", e);
-                onFailure.run();
+        ActionListener<PersistentTasksCustomMetadata.PersistentTask<?>> listener = ActionListener.wrap(
+            r -> logger.debug("Stopped geoip downloader task"),
+            e -> {
+                Throwable t = e;
+                if (e instanceof RemoteTransportException) {
+                    t = e.getCause();
+                }
+                if (t instanceof ResourceNotFoundException == false) {
+                    logger.error("failed to remove geoip downloader task", e);
+                    onFailure.run();
+                }
             }
-        });
+        );
         persistentTasksService.sendRemoveRequest(
             GEOIP_DOWNLOADER,
             ActionListener.runAfter(
                 listener,
                 () -> client.admin().indices().prepareDelete(DATABASES_INDEX).execute(ActionListener.wrap(rr -> {}, e -> {
-                    if (e instanceof ResourceNotFoundException == false) {
-                        logger.warn("failed to remove " + DATABASES_INDEX, e);
+                    Throwable t = e;
+                    if (e instanceof RemoteTransportException) {
+                        t = e.getCause();
+                    }
+                    if (t instanceof ResourceNotFoundException == false) {
+                        logger.warn(e.getClass() + "failed to remove " + DATABASES_INDEX, e);
```

Comment:
Nit: It seems we need a whitespace before `failed to remove` otherwise the class name won't be separated from it.

Reply:
It was left-over from debugging, I've removed `e.getClass()` entirely